### PR TITLE
[BE] Fix bugs

### PR DESCRIPTION
**Pull request by MihaiDruta**
[BE] Fix bugs
https://trello.com/c/NPEi8KKJ/95-be-fix-bugs

Fixed bugs:
Feedback:
- User is now able to give feedback only once to a specific user involved on an activity

Pagination:
- If the page is negative a HttpNotFoundException is thrown
- If the page exceeds the maximum number of pages available a HttpNotFoundException is thrown

### CODE REVIEW

File: src/Filters/PaginatorPageFieldValidator.php [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace App\Filters;
+
+class PaginatorPageFieldValidator
```
Comment (SpartakusMd) [inline src/Filters/PaginatorPageFieldValidator.php]: Maybe rename class to `PaginatorValidator` and the method to `isPageNumberValid`?

File: src/Repository/FeedbackRepository.php [hunk 2]
```diff
@@ -88,4 +89,12 @@ public function getUserFeedbackPaginated(
 
         return $query;
     }
+
+    public function canUserGiveFeedback(Activity $activity, User $userFrom, User $userTo): bool
```
Comment (SpartakusMd) [inline src/Repository/FeedbackRepository.php]: This method doesn't implement all the checks necessary, so it's name should be changed to `hasUserGivenFeedback()`

File: src/Handlers/UserHandler.php [hunk 3]
```diff
@@ -54,12 +63,17 @@ public function getUserListPaginated(
             $userListPagination->pageSize = $numResults;
         }
 
+        $numPages = (int)ceil($numResults / $userListPagination->pageSize);
+        if (!$this->parameterValidator->isParameterValid($numPages, $userListPagination->currentPage)) {
+            throw new NotFoundHttpException();
+        }
```
Comment (SpartakusMd) [inline src/Handlers/UserHandler.php]: The block `$context = ...` and `$json = $this->serializer...` should be after this validation. Check all the controllers.

File: src/Handlers/UserHandler.php [hunk 4]
```diff
@@ -54,12 +63,17 @@ public function getUserListPaginated(
             $userListPagination->pageSize = $numResults;
```
Comment (SpartakusMd) [inline src/Handlers/UserHandler.php]: Add this feature to all the paginated lists.

**Review by SpartakusMd — CHANGES_REQUESTED** [review]

File: src/Repository/ActivityRepository.php [hunk 5]
```diff
@@ -122,10 +122,13 @@ public function getPaginatedActivities(
         User $user
     ): Query {
         $queryBuilder = $this->getAvailableActivities($activityListSort, $activityListFilter, $user);
-
+        if ($activityListPagination->pageSize === -1) {
```
Comment (SpartakusMd) [inline src/Repository/ActivityRepository.php:138]: We can't have anymore `pageSize = -1` as we have `PaginatorValidator` :) Check all repozitories.

**Review by nspc97 — APPROVED** [review]

**Review by SpartakusMd — CHANGES_REQUESTED** [review]

**Review by SpartakusMd — APPROVED** [review]

**Comment by SpartakusMd** [comment]
@MihaiDruta please fix conflicts.

**Pull request merged**